internal/clients/ec2/ec2_client.go

```diff
@@ -99,3 +99,16 @@ func (c *Client) CreateEC2ClientFromConfig(crd *stsTypes.Credentials) (*Client,
 
 	return newClient, nil
 }
+
+func (c *Client) ListInstanceTypes() ([]types.InstanceTypeInfo, error) {
+	log.Trace().Msg("Listing AWS EC2 instance types")
+	input := &ec2.DescribeInstanceTypesInput{
+		// What is The maximum number of results to return for the request in a single page?
+	}
```

Comment:
The AWS API is designed to return between 0 - 100 items, the API returns items in random order and sorting or limit/offset is not possible. At some point, perhaps not in the initial version, we will hit the 100 limit in the API and we would need to:

* fetch results in multiple calls
* cache the result on the backend
* implement pagination in our API/UI

For the record, here is my community question and answers: https://repost.aws/questions/QUD4vDlkcRTBWVEg2gyC5_7w/sort-results-from-ec-2-api

We can probably skip this for types and hardcode an error if this returns more than 90 items (to be able to fix this later). Or if you want to go deep into this, let's implement multicalling now (unsure how to name this, its not pagination).

Reply:
Iteration is perhaps a good name for this, pagination for dealing with pages.